zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentReconstructProcessor.java

```diff
@@ -63,11 +70,17 @@ public DeploymentReconstructProcessor(
     formState = processingState.getFormState();
     decisionState = processingState.getDecisionState();
     stateWriter = writers.state();
+    commandWriter = writers.command();
     rejectionWriter = writers.rejection();
   }
 
   @Override
   public void processRecord(final TypedRecord<DeploymentRecord> record) {
+    if (record.getIntent() != DeploymentIntent.RECONSTRUCT) {
+      throw new IllegalArgumentException(
+          "Expected a record with intent DeploymentIntent.RECONSTRUCT, but got "
+              + record.getIntent());
+    }
```

Comment:
:upside_down_face: Technically doesn't hurt but it  looks a bit out of place here, no other processors check this, they leave it up to the processor registration to choose appropriate intents.